index.js

```diff
@@ -13,42 +19,47 @@ const defaultErrorResponder = (transformedErr, req, res, next) => next(transform
 
 const defaultKeysToValidate = ['params', 'body', 'query', 'user', 'headers', 'trailers', 'files']
 
-const middleware = (joiOptions, options) => (req, res, next) => {
+const middleware = (middlewareOptions = {}) => (req, res, next) => {
   // restify v7 uses req.route.spec
   const routeDefinition = req.route.spec || req.route
-  const {validation} = routeDefinition
+  const {schema, options = {}} = routeDefinition.validation || {}
 
-  if (!validation) {
+  if (!schema) {
     return setImmediate(next)
   }
 
-  const toValidate = options.keysToValidate.reduce((accum, key) => {
+  const joiOptions = options.joiOptions || middlewareOptions.joiOptions || defaultJoiOptions
+  const errorTransformer = options.errorTransformer || middlewareOptions.errorTransformer || defaultErrorTransformer
+  const errorResponder = options.errorResponder || middlewareOptions.errorResponder || defaultErrorResponder
+  const keysToValidate = options.keysToValidate || middlewareOptions.keysToValidate || defaultKeysToValidate
+
```

Comment:
The above is a bit awful to read, but it allows for route-level overrides and defaults